qt_binder/binder.py

```diff
@@ -568,6 +564,100 @@ def _on_trait_change(self, handler, name=None, remove=False,
+        overloads = defaultdict(list)
+        for i in range(meta_object.methodCount()):
+            meta_meth = meta_object.method(i)
+            qname = _python_name_for_qt_name(
+                _qt_name_for_meta_method(meta_meth))
+            name = renamings.get(qname, qname)
+            overloads[name].append(meta_meth)
+
+        for name, methods in overloads.items():
+            method_type = methods[0].methodType()
+            if method_type == QtCore.QMetaMethod.Slot:
+                trait_type = QtSlot
+            elif method_type == QtCore.QMetaMethod.Signal:
+                trait_type = QtSignal
+            else:
+                continue
+            for meta_meth in methods:
+                if len(methods) > 1:
+                    # Add the argument types to the name to disambiguate.
+                    arg_types = [_to_str(t).rstrip('*')
+                                 for t in meta_meth.parameterTypes()]
+                    qualname = '_'.join([name] + arg_types)
+                else:
+                    qualname = name
+                if qualname not in seen:
+                    if qualname.endswith('_'):
+                        # See #21
+                        continue
+                    trait = trait_type(meta_meth)
+                    binder_class.add_class_trait(qualname, trait)
+                    seen.add(qualname)
+
+    def _add_implied_properties(self, binder_class, renamings, seen):
+        """ Add properties defined by pairs of getters and setters.
+        """
+        # Add all getter/setter pairs that we can identify.
+        methods = set()
+        for name, class_attr in vars(binder_class.qclass).items():
+            # sip methoddescriptor objects do not have __call__() defined.
+            if (callable(class_attr) or
+                    type(class_attr).__name__ == 'methoddescriptor'):
+                methods.add(name)
+        for qname in methods:
+            # FIXME: We do not know if these are true getter/setters, where
+            # the getter has 0 arguments and the setter has just the 1. For
+            # example, `QObject.property(name)` and
+            # `QObject.setProperty(name, value)` get misidentified here.
+            # Unfortunately, the method objects do not have any information
+            # about their argument structure.
+            if qname == 'property':
+                continue
+            putative_setter = _setter_name(qname)
+            if putative_setter in methods:
+                name = renamings.get(qname, qname)
+                if name.endswith('_'):
+                    # See #21
+                    continue
+                if name in seen:
+                    # We've done this pair earlier, probably through the
+                    # QMetaProperty mechanism, so we can ignore it here.
+                    continue
+                binder_class.add_class_trait(
+                    name, QtGetterSetter(qname, putative_setter))
```

Comment:
this method adds the logic `if qname == 'property'` and `if name in seen` after moving lines 636-657 in here. The code also use `_setter_name` now instead of manually constructing the name by adding strings.